### PR TITLE
Language Detector C Tasks API and tests

### PR DESCRIPTION
**Pull request by kinaryml**

### CODE REVIEW

File: mediapipe/tasks/c/components/containers/language_detection_result.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  char* language_code;
+
+  float probability;
+};
+
+// Task output.
+struct LanguageDetectorResult {
+  struct LanguageDetectorPrediction* predictions;
+
+  // Keep the count of predictions.
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/components/containers/language_detection_result.h:38]: s/Keep the/The

File: mediapipe/tasks/c/components/containers/language_detection_result_converter.cc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include "mediapipe/tasks/c/components/containers/language_detection_result_converter.h"
+
+#include <cstdlib>
+
+#include "mediapipe/tasks/c/components/containers/language_detection_result.h"
+#include "mediapipe/tasks/cc/text/language_detector/language_detector.h"
+
+namespace mediapipe::tasks::c::components::containers {
+
+void CppConvertToLanguageDetectionResult(
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/components/containers/language_detection_result_converter.cc:25]: Please add a test for these APIs.

File: mediapipe/tasks/c/components/containers/language_detection_result.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+struct LanguageDetectorPrediction {
+  // An i18n language / locale code, e.g. "en" for English, "uz" for Uzbek,
+  // "ja"-Latn for Japanese (romaji).
+  char* language_code;
+
+  float probability;
+};
+
+// Task output.
+struct LanguageDetectorResult {
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/components/containers/language_detection_result.h:35]: These language specific results types should live in languate_detector.h

**Review by schmidt-sebastian — COMMENTED** [review]
Thank you. Some comments.

File: mediapipe/tasks/c/components/containers/language_detection_result_converter.cc (hunk 2 above)
Comment (kinaryml) [reply mediapipe/tasks/c/components/containers/language_detection_result_converter.cc:25]: I've added a test now. PTAL

File: mediapipe/tasks/c/components/containers/language_detection_result.h (hunk 3 above)
Comment (kinaryml) [reply mediapipe/tasks/c/components/containers/language_detection_result.h]: Fixed it but do let me know if `language_detection_result` or `language_detector_result` sounds better here.

mediapipe/tasks/c/components/containers/language_detection_result_converter.h
mediapipe/tasks/c/components/containers/language_detection_result_converter.cc

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**